src/transformers/models/zamba/modeling_zamba.py

```diff
@@ -0,0 +1,1728 @@
+                device=hidden_states.device,
+                dtype=dtype,
+            )
+            if attention_mask is not None and not torch.all(attention_mask == 1):
+                hidden_states = hidden_states * attention_mask.unsqueeze(1)
+            hidden_states = self.act(self.conv1d(hidden_states)[..., :seq_len])  # [batch, intermediate_size, seq_len]
+            if attention_mask is not None and not torch.all(attention_mask == 1):
+                hidden_states = hidden_states * attention_mask.unsqueeze(1)
+
+        # 3. State Space Model sequence transformation
+        # 3.a. Selection:  [batch, seq_len, self.time_step_rank + self.ssm_state_size * 2]
+        hidden_states = hidden_states.reshape(
+            -1, self.n_mamba_heads, self.intermediate_size // self.n_mamba_heads, seq_len
+        ).transpose(0, 1)
+        ssm_parameters = (self.x_proj_weight[:, None, :, :] @ hidden_states).transpose(-1, -2)
+
+        time_step, B, C = torch.split(
+            ssm_parameters, [self.time_step_rank, self.ssm_state_size, self.ssm_state_size], dim=-1
+        )
+        discrete_time_step = (self.dt_proj_weight[:, None] @ time_step.transpose(-1, -2)) + self.dt_proj_bias[
+            :, None, :, None
+        ]
+
+        discrete_time_step = nn.functional.softplus(discrete_time_step)
+
+        # 3.b. Discretization: B and C to [batch, seq_len, intermediate_size, ssm_state_size] (SRAM)
+        A = -torch.exp(self.A_log.float())
+        discrete_A = torch.exp(A[:, None, :, None, :] * discrete_time_step[:, :, :, :, None])  # [n_mamba_heads, batch, intermediate_size / n_mamba_heads, seq_len, ssm_state_size]
+        discrete_B = discrete_time_step[:, :, :, :, None] * B[:, :, None, :, :].float()  # [n_mamba_heads, batch, intermediate_size / n_mamba_heads, seq_len, ssm_state_size]
+        deltaB_u = discrete_B * hidden_states[:, :, :, :, None].float()  # [n_mamba_heads, batch, intermediate_size / n_mamba_heads, seq_len, ssm_state_size]
+        # 3.c perform the recurrence y ← SSM(A, B, C)(x)
+        scan_outputs = []
+        for i in range(seq_len):
+            ssm_state = discrete_A[:, :, :, i, :].transpose(0, 1) * ssm_state + deltaB_u[:, :, :, i, :].transpose(
+                0, 1
+            )  # [batch, n_mamba_heads, intermediate_size / n_mamba_heads, ssm_state_size]
+            scan_output = torch.matmul(ssm_state.transpose(0, 1).to(dtype), C[:, :, i, :].unsqueeze(-1))  # [n_mamba_heads, batch, intermediate_size / n_mamba_heads, 1]
+            scan_outputs.append(scan_output[:, :, :, 0])
+        scan_output = torch.stack(scan_outputs, dim=-1)  # [n_mamba_heads, batch, intermediate_size / n_mamba_heads, seq_len]
+        scan_output = scan_output + (hidden_states * self.D[:, None, :, None])  # [n_mamba_heads, batch, intermediate_size / n_mamba_heads, seq_len]
+        scan_output = scan_output * self.act(gate)
```

Comment:
`# fmt: off`  is needed at the top to keep comments on the right 😉 either remove the comments or indent them!

Reply:
dropped `# fmt: off` as well as the in-line comments